### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Wonsky1**
Implemented cache decorator

### CODE REVIEW

**Review by roffi37 — APPROVED** [review]

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,16 @@
...
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> int:
+        if args not in cached_results:
+            print("Calculating new result")
+            result = func(*args)
+            cached_results[args] = result
+            return result
```
Comment (fsocie7y) [inline app/main.py]: try to rewrite your decorator not to have `return` duplicated

**Review by fsocie7y — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (Wonsky1) [reply app/main.py]: @fsocie7y like the push i added last?

Comment (Wonsky1) [reply app/main.py]: Do i have to get rid of "else" block somehow, or it is fine?

**Review by fsocie7y — APPROVED** [review]
well done 🥇

**Pull request closed without merging**